tree/tree/src/TBasket.cxx

```diff
@@ -1003,18 +1031,34 @@ Int_t TBasket::WriteBuffer()
 
    // Transfer fEntryOffset table at the end of fBuffer.
    fLast = fBufferRef->Length();
-   if (fEntryOffset) {
-      // Note: We might want to investigate the compression gain if we
-      // transform the Offsets to fBuffer in entry length to optimize
-      // compression algorithm.  The aggregate gain on a (random) CMS files
-      // is around 5.5%. So the code could something like:
-      //      for(Int_t z = fNevBuf; z > 0; --z) {
-      //         if (fEntryOffset[z]) fEntryOffset[z] = fEntryOffset[z] - fEntryOffset[z-1];
-      //      }
-      fBufferRef->WriteArray(fEntryOffset,fNevBuf+1);
+   Int_t *entryOffset = GetEntryOffset();
+   if (entryOffset) {
+      Bool_t hasOffsetBit = fIOBits & static_cast<UChar_t>(TBasket::EIOBits::kGenerateOffsetMap);
+      if (!CanGenerateOffsetArray()) {
+         // If we have set the offset map flag, but cannot dynamically generate the map, then
+         // we should at least convert the offset array to a size array.
+         if (hasOffsetBit) {
+            for (Int_t idx = 1; idx < fNevBuf + 1; idx++) {
```

Comment:
Isn't it required to start from the last in order to actually get 'count' in the resulting array?

Reply:
Hm - yes, good catch!  Fixed.